.github/workflows/legacy-checkpoints.yml

```diff
@@ -26,15 +52,40 @@ jobs:
         aws-secret-access-key: ${{ secrets.AWS_SECRET_KEY_ID }}
         aws-region: us-east-1
 
-    - name: Generate checkpoint
+    - name: Install PL from source
+      working-directory: ./
+      env:
+        PACKAGE_NAME: pytorch
+      run: |
+        pip install -e . -f https://download.pytorch.org/whl/cpu/torch_stable.html
+        pip list
+      if: inputs.pl_version == ''
+
+    - name: Pull legacy checkpoints
+      working-directory: ./
+      run: bash .actions/pull_legacy_checkpoints.sh
+
+    - name: Generate checkpoints
       run: |
-        while IFS= read -r line; do
-            bash tests/legacy/generate_checkpoints.sh $line
-        done <<< $(cat tests/legacy/back-compatible-versions.txt)
+        bash generate_checkpoints.sh ${{ inputs.pl_version }}
+
+    - name: Upload checkpoints to GitHub Actions artifact
+      uses: actions/upload-artifact@v3
+      with:
+        name: checkpoints-${{ github.sha }}
+        path: tests/legacy/checkpoints/
+      if: always()
+
+    - name: Upload checkpoints to S3 (dryrun)
```

Comment:
I think I do because this workflow is not something triggered on every PR/push and I'd like to check what files get uploaded/overriden etc. with dryrun.